### PR TITLE
Add basic task support

### LINKED ISSUE #1935: Add vscode task provider like rls-vscode
I use this as my daily driver and it is awesome - thanks for providing such a great IDE experience!

Something that I miss from RLS (in Sublime & VSCode) was builtin tasks support.

Any chance of adding support for the [vscode task provider API](https://code.visualstudio.com/api/extension-guides/task-provider) like the official vscode extension has so that `cargo` tasks are automatically detected by vscode? Or is there a reason why [this](https://github.com/rust-lang/rls-vscode/blob/ffe36f31b7a7105c6127f44e407dd828f94d1709/src/tasks.ts) file wasn't cherry picked from [rust-lang/rls-vscode](https://github.com/rust-lang/rls-vscode)?

### PR DESCRIPTION
**Pull request by Timmmm**
This adds basic support for running `cargo build`, `cargo run`, etc.

Fixes #1935

I have tested this and it seems to work. There are two things I'm not sure about:

1. The workspace folder handling seems wrong - just get the first workspace folder? Is this just a TODO item? I don't know if it is right to lift `workspaceFolder` up to `activate()` but I couldn't see another way.
2. If you manually add an entry to `tasks.json` like this:

```
    {
      "type": "cargo",
      "command": "build",
      "problemMatcher": [
        "$rustc"
      ],
      "group": "build"
    }
```

then VSCode somehow magically knows to run `cargo build`. The documentation for `resolveTask` *sounds* like I should have to implement that for it to work:

```
 * Resolves a task that has no [`execution`](#Task.execution) set. Tasks are
 * often created from information found in the `tasks.json`-file. Such tasks miss
 * the information on how to execute them and a task provider must fill in
 * the missing information in the `resolveTask`-method.
```

But then it also says this:

```
* This method will not be
 * called for tasks returned from the above `provideTasks` method since those
 * tasks are always fully resolved. A valid default implementation for the
 * `resolveTask` method is to return `undefined`.
```

Either way, it works without implementing it so the only thing I can think is that it is doing some kind of crazy pattern matching of the tasks returned by `provideTasks()` and the ones found in `tasks.json`.

### CODE REVIEW

**Comment by edwin0cheng** [comment]
Just curious, do we want to implement these kind of features in server side, such that it could also be used in other editors ?

File: editors/code/src/tasks.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import {
+    Disposable,
+    ShellExecution,
+    Task,
+    TaskGroup,
+    TaskProvider,
+    tasks,
+    WorkspaceFolder,
+} from 'vscode';
```
[the remark this answers is not in the document]
Comment (Timmmm) [reply editors/code/src/tasks.ts]: Huh VSCode has a fixit to do this automatically! :-D

**Comment by matklad** [comment]
>Just curious, do we want to implement these kind of features in server side, such that it could also be used in other editors ?

We have **Rust Analyzer: Run** command, which is a strictly better feature. https://github.com/microsoft/language-server-protocol/issues/944 tracks its addition to the protocol (no one is actively working on that right now)

File: editors/code/src/client.ts [hunk 2]
```diff
@@ -30,15 +30,14 @@ export function configToServerOptions(config: Config) {
     };
 }
 
-export async function createClient(config: Config, serverPath: string): Promise<lc.LanguageClient> {
+export async function createClient(config: Config, serverPath: string, workspaceFolder: vscode.WorkspaceFolder): Promise<lc.LanguageClient> {
```
Comment (matklad) [inline editors/code/src/client.ts]:
```suggestion
export async function createClient(config: Config, serverPath: string, cwd: string): Promise<lc.LanguageClient> {
```

All we do with ws folder is extracting the path, so we as well might just accept the path in the first place.

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by matklad** [comment]
hm, 

bors r+

?

**Pull request merged**